components/datetime/src/provider/date_time.rs

```diff
@@ -158,23 +159,29 @@ where
         time_length: length::Time,
         preferences: Option<preferences::Bag>,
     ) -> Result<DataPayload<PatternPluralsFromPatternsV1Marker>> {
-        let patterns_data = self.patterns_data_payload()?;
-        patterns_data.try_map_project_with_capture(
-            (date_length, time_length, preferences),
-            |data, (date_length, time_length, preferences), _| {
+        let time_patterns_data = self.time_patterns_data_payload()?;
+
+        let date_patterns_data = self.date_patterns_data_payload()?;
+        date_patterns_data.try_map_project_with_capture(
+            (date_length, time_length, preferences, time_patterns_data),
+            |data, (date_length, time_length, preferences, time_patterns_data), _| {
                 let date = pattern_for_date_length_inner(data.clone(), date_length)
                     .expect_pattern("Lengths are single patterns");
-                let time = pattern_for_time_length_inner(data.clone(), time_length, &preferences)
-                    .expect_pattern("Lengths are single patterns");
 
                 let pattern = match date_length {
                     length::Date::Full => data.length_combinations.full,
                     length::Date::Long => data.length_combinations.long,
                     length::Date::Medium => data.length_combinations.medium,
                     length::Date::Short => data.length_combinations.short,
                 };
-                let pattern = pattern.combined(date, time)?;
-                Ok(PatternPlurals::from(pattern).into())
+
+                let time = pattern_for_time_length_inner(
+                    time_patterns_data.get().clone(),
```

Comment:
Thought: I think #1131 should address this extra clone.

Nit: Can you add a TODO pointing at #1131 to get rid of the clone?